shm/Player.hpp

```diff
@@ -0,0 +1,22 @@
+#pragma once
+#include <memory>
```

Comment:
Ship.hpp has memory,so you do not need it here.

Reply:
+1